sdk/storage/azure-storage-blob/azure/storage/blob/_shared/uploads_async.py

```diff
@@ -421,3 +423,51 @@ async def _upload_chunk(self, chunk_offset, chunk_data):
     # TODO: Implement this method.
     async def _upload_substream_block(self, index, block_stream):
         pass
+
+
+class AsyncIterStreamer():
+    """
+    File-like streaming object for AsyncGenerators.
+    """
+    def __init__(self, generator: AsyncGenerator[Union[bytes, str], None], encoding: str = "UTF-8"):
+        self.generator = generator
+        self.iterator = generator.__aiter__()
+        self.leftover = b""
+        self.encoding = encoding
+
+    def __len__(self):
+        return self.generator.__len__()
+
+    def __aiter__(self):
```

Comment:
@annatisch, curious if you think we actually need to include `__aiter__` and `__anext__` here. I did this because the existing sync `IterStreamer` does but the idea behind this class is to wrap a generator in a stream so I think we only use stream methods from this class. I don't see why we would also need this class to be an Iterable.